src/pages/costModels/costModel/updateMarkupDialog.tsx

```diff
@@ -62,76 +91,111 @@ class UpdateMarkupModelBase extends React.Component<Props, State> {
-                      if (!isNaN(markupDecimal) && dx > -1 && markup.length - dx - 1 > 2) {
-                        this.setState({
-                          markup: formatValue(markupDecimal, 'markup', {
-                            fractionDigits: 2,
-                          }) as string,
-                        });
-                        return;
-                      }
-                      this.setState({ markup });
-                    }}
-                    validated={!isNaN(Number(this.state.markup)) ? 'default' : 'error'}
-                  />
-                  <InputGroupText style={{ borderLeft: '0' }}>%</InputGroupText>
-                </InputGroup>
-              </FormGroup>
+              <Grid hasGutter>
+                <GridItem lg={8} id="refSign">
+                  <FormGroup isInline fieldId="markup-or-discount" label={t('cost_models_details.markup_or_discount')}>
+                    <div style={styles.radioAlign}>
+                      <Radio
+                        isChecked={this.state.sign === '+'}
+                        name="markup"
+                        label={t('cost_models_details.markup_plus')}
+                        aria-label={t('cost_models_details.markup_plus')}
+                        id="markup"
+                        value="+"
+                        onChange={this.handleSignChange}
+                      />
+                    </div>
+                    <div style={styles.radioAlign}>
+                      <Radio
+                        isChecked={this.state.sign === '-'}
+                        name="discount"
+                        label={t('cost_models_details.discount_minus')}
+                        aria-label={t('cost_models_details.discount_minus')}
+                        id="discount"
+                        value="-"
+                        onChange={this.handleSignChange}
+                      />
+                    </div>
+                  </FormGroup>
+                </GridItem>
+                <GridItem lg={4} id="refMarkup">
+                  <FormGroup
+                    isInline
+                    fieldId="rate"
+                    label={t('rate')}
+                    helperTextInvalid={t('cost_models_wizard.markup.invalid_markup_text')}
+                  >
+                    <InputGroup style={{ width: '150px' }}>
+                      <InputGroupText style={{ borderRight: '0' }}>{this.state.sign}</InputGroupText>
+                      <TextInput
+                        style={{ borderLeft: '0' }}
+                        type="text"
+                        aria-label={t('rate')}
+                        id="markup-input-box"
+                        value={this.state.markup}
+                        onChange={this.handleMarkupDiscountChange}
+                        validated={this.markupValidator()}
+                      />
+                      <InputGroupText style={{ borderLeft: '0' }}>%</InputGroupText>
```

Comment:
It would be best to create a class for `borderLeft: '0'` and `borderRight: '0'`. Try to avoid inline styles

Reply:
Agreed, just missed that.